### ISSUE TITLE
ZIO.foreach and friends for a NonEmptyList

### ISSUE THREAD

**Issue by hmemcpy** [zio/interop-cats#315]
It would be great if this project could provide an extension for ZIO that added `ZIO.foreach/collectAll` etc for Cats NEL.

**Comment by neko-kai** [comment]
It doesn't seem like it can be done because implicit class search only happens if the method with the same __name__ does not exist on the class, even if the type of the argument doesn't match. Example: https://scastie.scala-lang.org/6Qdbo7PhQwKkA2oOo2VM3g

One weird thing that could enable this is providing implicit conversion `NonEmptyList[A] => Iterable[A]` since ZIO.foreach works with any iterable - but then you wouldn't get a Nel back as a result.

You could also just use cats' `.traverse` syntax...

**Comment by hmemcpy** [comment]
Yeah, I thought about adding the implicit conversion to Iterable. I didn't realize the extension method trick wouldn't work... And we do use traverse for those places that expose NELs, but it would be nice having a consistency with ZIO. Anyway, thanks, I'll try seeing if

**Comment by adamgfraser** [comment]
Perhaps we could provide it under a different name? `foreachNonEmptyList`? I ran into this too.

**Comment by neko-kai** [comment]
If that would work for @hmemcpy too let's add a `foreachNonEmptyList`/`foreachNel`/etc extension method

**Comment by joroKr21** [comment]
If you are using Cats / Cats Effect you can just do `nel.traverse` etc.
